tests/helpers.js

```diff
@@ -30,9 +32,14 @@ export class ZipFile {
    *
    * The onRead callback receives a single argument, a yauzl Entry object.
    */
-  readEach(onRead) {
+  readEach(onRead: Function): Promise {
     return new Promise((resolve, reject) => {
 
+      if (!this._zip) {
+        throw new Error(
```

Comment:
Currently when this error is thrown (it shouldn't, but if it happens) the test case which use this helpers (tests/test-cmd/test-build.js) fails to detect the error and it fails for timeout reasons instead of showing the error throwed in the failure logs.

I propose to tweak the test case into something like the following diff:

```
diff --git a/tests/test-cmd/test.build.js b/tests/test-cmd/test.build.js
index ef19418..d5eff39 100644
--- a/tests/test-cmd/test.build.js
+++ b/tests/test-cmd/test.build.js
@@ -28,21 +28,27 @@ describe('build', () => {
                          /minimal_extension-1\.0\.xpi$/);
             return buildResult.extensionPath;
           })
           .then((extensionPath) => zipFile.open(extensionPath))
           .then(() => {
             var fileNames = [];
-            return new Promise((resolve) => {
+            return new Promise((resolve, reject) => {
               zipFile.readEach((entry) => {
                 fileNames.push(entry.fileName);
               })
               .then(() => {
                 resolve(fileNames);
+              })
+              .catch((error) => {
+                reject(error);
               });
             });
           })
           .then((fileNames) => {
             assert.deepEqual(fileNames, ['manifest.json']);
           })
+          .catch((error) => {
+            assert.ifError(error);
+          })
       );
     });
```